query/graphql/src/main/java/org/codice/ddf/admin/graphql/transform/FunctionDataFetcherException.java

```diff
@@ -14,19 +14,22 @@
 package org.codice.ddf.admin.graphql.transform;
 
 import com.google.common.collect.ImmutableMap;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
-import org.boon.Boon;
 import org.codice.ddf.admin.api.report.ErrorMessage;
 
 public class FunctionDataFetcherException extends RuntimeException {
 
   private final ArrayList<ErrorMessage> customMessages;
+  private static final Gson GSON_PP =
```

Comment:
❓ Is this threadsafe? If multiple threads construct this exception concurrently can they step on each other? Would there be a big performance hit creating a new `Gson` or a new `GsonBuilder` + `Gson` each time through instead?

Reply:
Found this: https://stackoverflow.com/questions/10380835/is-it-ok-to-use-gson-instance-as-a-static-field-in-a-model-bean-reuse

Reply:
I'll look into if this is thread safe, the problem was not with performance but rather that `super` needs to be called before `GSON_PP` is referenced. However, since it's passed into the super constructor I had to make `GSON_PP` static to please the compiler.